### PR TITLE
Create MockTextRenderer

### PR DESCRIPTION
**Pull request by karupayun**
In this PR we are creating MockTextRenderer. Similar to MockBatcher, it will 
be used by CaptureViewElements to test UpdatePrimitives and DrawMethods().

- I also changed all the rectangle methods to use (top_left and size).

Bug: http://b/228609901.

### CODE REVIEW

File: src/OrbitGl/CoreMath.h [hunk 1]
```diff
@@ -24,4 +24,17 @@ using Color = gte::Vector4<unsigned char>;
 [[nodiscard]] inline Vec2 Vec3ToVec2(const Vec3& v) { return {v[0], v[1]}; }
 [[nodiscard]] inline Vec3 Vec2ToVec3(Vec2 vertex, float z = 0) { return {vertex[0], vertex[1], z}; }
 
+[[nodiscard]] inline bool IsInBetween(float point, float min, float max) {
+  return point >= min && point <= max;
+}
+
+[[nodiscard]] inline bool IsInsideRectangle(Vec2 point, Vec2 top_left, Vec2 size) {
```
Comment (antonrohr) [inline src/OrbitGl/CoreMath.h:31]: maybe const ref for all parameters?

File: src/OrbitGl/CoreMath.h [hunk 2]
```diff
@@ -24,4 +24,17 @@ using Color = gte::Vector4<unsigned char>;
...
+  for (int i = 0; i < 2; i++) {
+    if (!IsInBetween(point[i], top_left[i], top_left[i] + size[i])) {
+      return false;
+    }
+  }
+  return true;
+}
```
Comment (beckerhe) [inline src/OrbitGl/CoreMath.h:38]: Can you put these into the orbit_gl namespace?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/OrbitGl/MockTextRenderer.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    std::numeric_limits<float>::max()};
+  max_point_ = Vec3{std::numeric_limits<float>::lowest(), std::numeric_limits<float>::lowest(),
+                    std::numeric_limits<float>::lowest()};
+  num_characters_in_add_text_.clear();
+  vertical_position_in_add_text.clear();
+  num_add_text_calls_ = 0;
+}
+
+void MockTextRenderer::AddText(const char* text, float x, float y, float z,
+                               TextFormatting formatting) {
```
Comment (karupayun) [inline src/OrbitGl/MockTextRenderer.cpp:25]: TODO: Consider formatting in the final x, y positions.

File: src/OrbitGl/MockTextRenderer.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+// GetStringHeight is clearly over-estimated here. Comparting to the one in OpenGlTextRenderer, the
+// real height is 10 (with font-size 14) and 8 (with font-size 10). It should not be a problem.
+float MockTextRenderer::GetStringHeight(const char* /*text*/, uint32_t font_size) {
+  return font_size;
+}
+
+[[nodiscard]] bool MockTextRenderer::IsTextInsideRectangle(Vec2 start, Vec2 size) const {
+  return IsInsideRectangle({min_point_[0], min_point_[1]}, start, size) &&
```
Comment (antonrohr) [inline src/OrbitGl/MockTextRenderer.cpp:60]: Again, I prefer to use `Vec3ToVec2`

File: src/OrbitGl/MockTextRenderer.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  float AddTextTrailingCharsPrioritized(const char* text, float x, float y, float z,
+                                        TextFormatting formatting,
+                                        size_t trailing_chars_length) override;
+
+  [[nodiscard]] float GetStringWidth(const char* text, uint32_t font_size) override;
+  [[nodiscard]] float GetStringHeight(const char* text, uint32_t font_size) override;
+
+  bool HasAddTextsSameLength() const { return num_characters_in_add_text_.size() <= 1; }
+  bool AreAddTextsAlignedVertically() const { return vertical_position_in_add_text.size() <= 1; }
+  const int GetNumAddTextCalls() const { return num_add_text_calls_; }
```
Comment (antonrohr) [inline src/OrbitGl/MockTextRenderer.h:38]: no discard?

**Review by antonrohr — APPROVED** [review]
Again, this might seem a bit trivial, but I prefer to have a test for every file, in this case CoreMath.h. 
Besides the actual testing reason, there are also 2 other benefits:
1. Whoever touches CoreMath.h next and implements a less trivial function, is reminded to add a test, because a test file already exists.
2. It defines the behaviour of the functions in a way and serves as documentation. `IsInBetween` currently uses `>=` and `<=`. One edge case test then is `EXPECT_TRUE(3,3,3)` and therefore defines the exact behaviour of the function

File: src/OrbitGl/CoreMath.h [hunk 6]
```diff
@@ -24,4 +24,17 @@ using Color = gte::Vector4<unsigned char>;
 [[nodiscard]] inline Vec2 Vec3ToVec2(const Vec3& v) { return {v[0], v[1]}; }
 [[nodiscard]] inline Vec3 Vec2ToVec3(Vec2 vertex, float z = 0) { return {vertex[0], vertex[1], z}; }
 
+[[nodiscard]] inline bool IsInBetween(float point, float min, float max) {
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/CoreMath.h:27]: Agree with value, suggestion about naming?

[the remark this answers is not in the document]
Comment (antonrohr) [reply src/OrbitGl/CoreMath.h:27]: Not really.. I thought maybe there are good names from other programming languages/frameworks, but I didn't really find anything. From a maths dircection, maybe `IsInInterval` makes sense.. or `IsInBounds`. I dont have a strong opinion and `IsInBetween` is also fine

[the remark this answers is not in the document]
Comment (beckerhe) [reply src/OrbitGl/CoreMath.h:27]: These functions always have the problem that they don't properly define whether they operate on open, half-open or closed intervals. That's one of the big reasons you won't find them in standard libraries.

The proper solution would be to define a type: `ClosedInterval` and define the function as `bool IsElementOf(float value, ClosedInterval interval)`. In this way it can be overloaded later on for other types - like `OpenInterval` or whatever can fulfills the `IsElementOf` operation.

[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/CoreMath.h]: What do you think now?

**Comment by karupayun** [comment]
I made several changes including taking care of alignment in MockTextRenderer::AddText, si I'm asking for re-review.

File: src/OrbitGl/CoreMath.h [hunk 7]
```diff
@@ -21,20 +21,34 @@ using Vec4i = gte::Vector4<int>;
...
 [[nodiscard]] inline Vec3 Vec2ToVec3(Vec2 vertex, float z = 0) { return {vertex[0], vertex[1], z}; }
 
-[[nodiscard]] inline bool IsInBetween(float point, float min, float max) {
-  return point >= min && point <= max;
+namespace orbit_gl {
+
+struct ClosedInterval {
+  static ClosedInterval FromValues(float value_1, float value_2) {
+    return {std::min(value_1, value_2), std::max(value_1, value_2)};
+  }
```
Comment (antonrohr) [inline src/OrbitGl/CoreMath.h:33]: I dont understand why this "complicated" method. AFAIK, wherever you use this method, it is already know which value is min (always value_1) and which is max (always value_2). 

I would do a simple explicit constructor.

**Review by antonrohr — APPROVED** [review]
LGTM only one comment

File: src/OrbitGl/CoreMath.h (hunk 7 above)
Comment (antonrohr) [reply src/OrbitGl/CoreMath.h:33]: Discussed offline
